src/main/java/com/hivemq/cli/commands/options/TlsOptions.java

```
@@ -138,104 +127,255 @@ private boolean useBuiltSslConfig() {
+        final boolean defaultCertificateExists = defaultServerCertificate != null;
+
+        if (truststoreExists) {
+            if (certificatesExists) {
+                Log.warn("The truststore parameter shadows the server/ca certificate parameter. " +
+                        "If you want to use them instead please remove the keystore parameter.");
+            }
+            trustManagerFactory =
+                    TlsUtil.createTrustManagerFactoryFromTruststore(clientTruststorePath, clientTruststorePassword);
+        } else if (certificatesExists) {
+            final KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+
+            final Collection<X509Certificate> certificates = new ArrayList<>();
+            if (serverCertificatePath != null) {
+                certificates.addAll(TlsUtil.getCertificateChainFromFile(serverCertificatePath));
+            }
+            if (serverCertificateDirPath != null) {
+                certificates.addAll(TlsUtil.getCertificateChainFromDirectory(serverCertificateDirPath));
+            }
+            if (!certificates.isEmpty()) {
+                keyStore.load(null, null);
+
+                int i = 1;
+                for (final X509Certificate cert : certificates) {
+                    final String alias = "File_Certificate_" + i;
+                    keyStore.setCertificateEntry(alias, cert);
+                    i++;
+                }
+            }
+            trustManagerFactory = TrustManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init(keyStore);
+        } else if (defaultTruststoreExists) {
+            if (defaultCertificateExists) {
+                Log.warn("The truststore default option (properties file) shadows the server/ca " +
+                        "certificate default option (properties file). " +
+                        "If you want to use them instead please remove the keystore default option.");
```

I would throw an Exception here.